### PR TITLE
Add Internship Tag

### LINKED ISSUE #56: [TASK] Add Tags component for the internships
## Description
We should add a `InternshipTag.jsx` component that will accept an icon `name` and `label` as props and render the tag to look like below

Image

## Related
https://github.com/BobaTalks/internship-tracker/issues/25

### PR DESCRIPTION
**Pull request by aqvilala**
#### What is the purpose of your change?
- Fixes #56 

#### What areas of the application does this change impact? (changes API, folder structure, etc)
- Adds a new component 

#### What features/fixes are deferred/not handled in this PR?
- N/A

## Checklist
- [ ] Adds unit tests
- [ ] Has a work item linked to the PR

### CODE REVIEW

File: client/src/components/InternshipTag.jsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        sx={{
+          borderRadius: "0.5rem",
+          padding: "0.5rem 1rem",
+          bgcolor: theme.palette.blue.lighter,
+          color: theme.palette.text.primary,
+          border: 1,
+          maxWidth: "fit-content",
+        }}
+      >
+        <props.icon />
```
Comment (jwu910) [inline client/src/components/InternshipTag.jsx:21]: If we're passing an element into the props for this, we could destructure it from props first and then render it with the <> like a component

```jsx
const { icon: Icon } = props
// or 
const Icon = props.icon

// and then 

<Box>
  <Icon />
</Box> 
```

File: client/src/components/InternshipTag.jsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import React, { useState } from "react";
+import { Box, useTheme } from "@mui/material";
+
+const InternshipTag = (props) => {
+  let theme = useTheme();
+
+  return (
+    <>
```
Comment (jwu910) [inline client/src/components/InternshipTag.jsx:8]: If we're returning a single element `Box` here, the fragment is unnecessary and we can remove it.

**Review by jwu910 — CHANGES_REQUESTED** [review]
nice job on this PR, just two changes to make and we could be good to go. Are there screenshots of this from your local testing we can take a look at as well?

**Comment by jwu910** [comment]
Let us know if you need any support with this @aqvilala

File: client/src/components/InternshipTag.jsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import React, { useState } from "react";
+import { Box, useTheme } from "@mui/material";
+
+const InternshipTag = (props) => {
+  let theme = useTheme();
```
Comment (michachan) [inline client/src/components/InternshipTag.jsx:5]: prefer const for variables not redefined

**Comment by jwu910** [comment]
@aqvilala Checking in again, any support we can provide with this work? If you need to pass this on, let us know so we can keep it moving. Thanks

**Comment by aqvilala** [comment]
image

Here's what the tag looks like on my local! I've made the changes and updated the PR

**Review by jwu910 — APPROVED** [review]
Nice, thanks!

**Pull request merged**